pages/learn/deploy/offline-updates.md

```diff
@@ -0,0 +1,273 @@
+
+If this is the case, we recommend mounting an external mass storage (USB) device into a privileged data container and sharing it to other containers (if applicable) via NFS or similar network storage protocol. These external storage devices are not part of the upgrade process, so data on them would be left intact, as long as they are temporarily disconnected during the upgrade process. By contrast, a typical balena online update leaves services and data intact, and persistent logging can be enabled to save your logs across device restarts.s
+
+## Performing an offline upgrade 
+
+To perform an offline upgrade, we will be using [balena-cli](https://www.balena.io/docs/reference/balena-cli/). All commands should work on Linux distributions running Docker on Linux Kernel with AUFS and overlay filesystem support. For Windows and macOS, the last version of Docker Desktop supporting AUFS is 18.06.1
+
+<A line about Mention how important env variables in the process>
+<A line about optional stages and where actually the process is the same for both new and existing devices>
+
+> Note: In an offline upgrades process, all data on device is wiped at this point, making this different from a typical software update process in the balena ecosystem. In this process, any additional user data and system settings written to various device partitions would be lost.
+
+Offline updates includes the following steps:
+
+- [Setup](#setup)
+- [Create application](#createuse-pre-existing-application)
+- [Create offline device](#createuse-pre-existing-offline-device)
+- [Download balenaOS image](#download-balenaos-image)
+- [Configure balenaOS image](#configure-balenaos-image)
+- [Create and preload release](#create-and-preload-release)
+- [Create update media](#create-update-media)
+- [Reprovisioning process](#process-of-reprovisioning)
+- [Update device registration(s)](#update-device-registrations)
+
+
+### Setup
+
+> Download and install [balena-cli][balena-cli] on a linux distribution. The commands have been tested to work on Ubuntu 20.04.
+
+Several `balena-cli` commands require access to a balenaCloud account. Those commands require creating a CLI login session by running [`balena login`](https://www.balena.io/docs/reference/balena-cli/#logging-in) in the terminal. 
+
+```bash
+$ balena login
+```
+
+Store the filename of your OpenSSH private key in an environment variable or generate an SSH key using the commands below. The SSH keys will be added to the device and will be used to SSH into the device after provisioning that will help 
+
+```bash
+# (optional) if you already have SSH key(s)
+$ ssh-keygen -o -a 100 -t ed25519 -f id_ed25519 -C 'offline-updates' -N ''
+
+# must contain the filename of the OpenSSH private key
+$ ssh_key= id_ed25519.pub
```

Comment:
@vipulgupta2048 there should be no space, this command is invalid otherwise.

Reply:
Thanks @ab77 Good catch